modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/security/permission/propagator/DLFolderPermissionPropagatorImpl.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.web.internal.security.permission.propagator;
+
+import com.liferay.document.library.constants.DLPortletKeys;
+import com.liferay.document.library.kernel.model.DLFolder;
+import com.liferay.document.library.kernel.service.DLFolderLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.security.permission.propagator.BasePermissionPropagator;
+import com.liferay.portal.kernel.security.permission.propagator.PermissionPropagator;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+import javax.portlet.ActionRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Loc Pham
+ */
+@Component(
+	property = "javax.portlet.name=" + DLPortletKeys.DOCUMENT_LIBRARY_ADMIN,
+	service = PermissionPropagator.class
+)
+public class DLFolderPermissionPropagatorImpl extends BasePermissionPropagator {
+
+	@Override
+	public void propagateRolePermissions(
+		ActionRequest actionRequest, String className, String primKey,
+		long[] roleIds) {
+	}
+
+	@Override
+	public void propagateRolePermissions(
+			long groupId, long companyId, String className, String primKey,
+			String childClassName, String childPrimKey, long[] roleIds)
+		throws PortalException {
+
+		DLFolder dlFolder = _dlFolderLocalService.fetchFolder(
+			GetterUtil.getLong(childPrimKey));
+
+		if (dlFolder == null) {
+			return;
+		}
```

Comment:
I think letting it throw an Exception here is better than checking for null. What do you think?

Reply:
If not want check null, it should use the get function

Reply:
No need. The exception will be thrown in propagateRolePermissions() if there's no folder with primKey.